File: Day01/day01.cs
using System;
using System.IO;

namespace AoC2019
{
    class Program
    {
        static void Main (string[] args)
        {
            string inputPath = @"input";
            if (!File.Exists(inputPath))
            {
                Console.WriteLine($"Error: input file {inputPath} not found");
                return;
            }

            Console.WriteLine("Day 01");
            using (StreamReader reader = new StreamReader(inputPath))
            {
                long sum = 0;
                string line;
                // Read lines 'by hand', because ReadToEnd().Split("\n") adds an entry that breaks
                // Linq
                while (null != (line = reader.ReadLine()))
                {
                    sum += Int32.Parse(line) / 3 - 2;
                }
                Console.WriteLine($"1. Total Fuel: {sum}");
            }
        }
    }
}

File: Day01/day01.py
#!/usr/bin/env python3

def sum_fuel (mass):
    fuel = mass // 3 - 2
    total = 0
    while fuel > 0:
        total += fuel
        fuel = fuel // 3 - 2
    return total

# Read data
with open("input") as f:
    data = f.readlines()

# Part 1
fuel = sum([int(n) // 3 - 2 for n in data])
print(f"1. Total Fuel: {fuel}")

# Part 2
fuel = sum([sum_fuel(int(n)) for n in data])
print(f"2. Total Fuel: {fuel}")